streams/shared/src/main/scala/scalaz/stream/ZStream.scala

```diff
@@ -660,6 +660,38 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
         }
       }
   }
+
+  /**
+   * Stores this stream to a fixed bounded queue and streams data from the bounded queue.
+   */
+  final def buffer[R1 <: R, E1 >: E](capacity: Int = 1): ZStream[R1, E1, A] =
```

Comment:
I suggest not using a default value for capacity and forcing the user to explicitly choose one. Also, the scaladoc should recommend a power-of-two capacity so the underlying queue would use the more efficient implementation.

Reply:
Fixed